### PR TITLE
Bring back legacy verification by comparing public device keys

### PR DESCRIPTION
**Pull request by bwindels**
Fixes https://github.com/vector-im/riot-web/issues/12621
There is no design yet for this, but as a few things are blocked behind design, I wanted to have a go anyway hoping any design corrections will be minimal afterwards.

legacyverif

@nadonomy is this good enough for now?

### CODE REVIEW

**Comment by nadonomy** [comment]
@bwindels principally yes, as the UX matches what we have & want in Figma and our minutes from the last time we discussed legacy verification.

The copy could do with improving though, which I can spend time on this morning. Would you prefer to block this PR for a few more hours while I gather that, or merge as is & we handle that separately?

**Comment by bwindels** [comment]
> @bwindels principally yes, as the UX matches what we have & want in Figma and our minutes from the last time we discussed legacy verification.
> 
> The copy could do with improving though, which I can spend time on this morning. Would you prefer to block this PR for a few more hours while I gather that, or merge as is & we handle that separately?

Sure, sounds good! Fwiw, I couldn't spot any design for this in figma.

File: docs/builtins.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+```
+const url = new URL("https://foo.com/path?foo=bar");
+url.hostname; // foo.com
+url.pathname; // /path
+url.searchParams.get("foo"); //bar
+```
+
+## Internationalization
+
+`Intl` can do a lot nowadays
```
Comment (dbkr) [inline docs/builtins.md]: Was this file intentional? The URL bit looks pretty useful but guessing the intl bit is WIP!

File: src/components/views/dialogs/ManualDeviceKeyVerificationDialog.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2016 OpenMarket Ltd
+Copyright 2017 Vector Creations Ltd
+Copyright 2019 New Vector Ltd
+Copyright 2019 Michael Telatynski <7t3chguy@gmail.com>
```
Comment (dbkr) [inline src/components/views/dialogs/ManualDeviceKeyVerificationDialog.js]: need to add a 2020?

File: docs/builtins.md (hunk 1 above)
Comment (bwindels) [reply docs/builtins.md]: Indeed it is :)

**Comment by nadonomy** [comment]
@bwindels for now to unblock this PR can we update the copy as following please?

Screenshot 2020-03-20 at 10 53 43

- [ ] Relabel [Legacy Verify Device] to [Manually Verify by Text]
- [ ] Relabel [Verify Device by Emoji] to [Interactively verify by Emoji]

Screenshot 2020-03-20 at 10 53 46

Update copy to:
- [ ] Update first paragraph to: Confirm this users session by comparing the following with their User Settings:
- [ ] Update second paragraph to: If they don't match, the security of your communication may be compromised.
- [ ] Relabel [I verify the keys match] to [Verify session]

**Comment by bwindels** [comment]
Sorry for the force push, wanted to remove the file I didn't meant to commit from the history for now.
